### PR TITLE
Add error tracking via Sentry

### LINKED ISSUE #1902: Add error tracking
We should have at least some error tracking in place. Of course this is a static page which little potential for things to go wrong and 99% of the JS is optional anyway really but it would be good to have a feeling for whether things generally work or not. One specific case where we'd absolutely need to know if errors happen is the contact form – if that doesn't submit successfully, we should be aware.

## TODOs

- [x] add general Sentry catchall code for JS
- [x] report submission errors of the contact form to Sentry explicitly (including the original message so that isn't lost)
- [x] add a clause about usage of Sentry to the [privacy policy](https://mainmatter.com/privacy/)
- [x] enable source maps for JS (as we'll need them for Sentry)

### PR DESCRIPTION
**Pull request by marcoow**
This adds Sentry:

* enables sourcemaps for more meaningful logs on Sentry
* tracks any uncaught errors with Sentry
* explicitly logs failed contact form deliveries to Sentry
* adds Sentry to our privacy policy

closes #1902

### CODE REVIEW

File: src/assets/js/app.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+if (host === "mainmatter.com") {
+  environment = "production";
+} else if (host.endsWith("netlify.app")) {
+  environment = "staging";
+}
+
+Sentry.init({
+  dsn: "https://43f7627909d94dc4a769340ad730f1a2@o68744.ingest.sentry.io/4504039028817920",
+  environment
+});
```
Comment (Turbo87) [inline src/assets/js/app.js]: sure that we want to enable it in development mode too? Sentry by default shows all environments and seeing devtime errors in there could be a bit confusing.

File: src/privacy.njk [hunk 2]
```diff
@@ -488,7 +488,22 @@ noindex: true
...
+      ensure a smooth experience for all users.
+    </p>
+    <p>
+      The operator of the Sentry component is Functional Software Inc., 45 Fremont Street 8th Floor, San Francisco, CA 94105, United States.
+    </p>
+    <p>
+      For Sentry's policies and practices around analysis and storage of user data, refer to <a href="https://sentry.io/legal">https://sentry.io/legal</a>.
+    </p>
+    <h2>
+      10. Data protection provisions about the application and use of Google Analytics (with anonymization function)
```
Comment (Turbo87) [inline src/privacy.njk]: seems unrelated?

File: package.json [hunk 3]
```diff
@@ -63,6 +63,8 @@
     "last 1 Edge versions"
   ],
   "dependencies": {
+    "@sentry/browser": "^7.16.0",
+    "@sentry/tracing": "^7.16.0",
```
Comment (Turbo87) [inline package.json]: this does not seem to be used anywhere?

File: src/privacy.njk (hunk 2 above)
Comment (Turbo87) [reply src/privacy.njk]: nevermind, I had mistaken this for the changes in the other two PRs 🤦

File: package.json (hunk 3 above)
Comment (marcoow) [reply package.json]: ah. right – should be removed 👍

File: src/assets/js/app.js (hunk 1 above)
Comment (marcoow) [reply src/assets/js/app.js]: fair point – let's enable this on mainmatter.com only

**Review by BobrImperator — APPROVED** [review]
